### PR TITLE
feat(ledger): updated ledger signing to work with Generic and Migration

### PR DESCRIPTION
**Pull request by bee344**
Update the signing options that use Ledger to work with both the Generic App or the Migration App.

Notes:

- `getLedgerAddress()` now only works with the Ledger Generic App, it could be made to work with both with quite a bit of changes, but don't know if it's worth it
- I removed `@zondax/ledger-substrate` and `@ledgerhq/hw-transport-node-hid` since I couldn't find where they were used.

TODO:

- [x] More testing

### CODE REVIEW

File: blockchainServices/setup.ts [hunk 1]
```diff
@@ -131,18 +184,21 @@ const ledgerSignAndSend = async (call: any, api: any) => {
...
 
 	const network = await getNetwork();
+
+	const ss58 = api.consts.system.ss58Prefix.toNumber();
 	
-	const ledger = new Ledger('hid', network);
-	const sender = await ledger.getAddress();
+	const ledger = migration ? new LedgerGeneric('hid', network, knownLedger[network]) : new LedgerGeneric('hid', network, knownLedger['polkadot']);
+	const sender = await ledger.getAddress(ss58, false, ledger.accountOffset, ledger.addressOffset);;
 	console.log({ sender });
```
Comment (TarikGul) [inline blockchainServices/setup.ts:196]: Should we remove this `console.log`. I know it's not part of the PR, but I just noticed it.

File: blockchainServices/setup.ts [hunk 2]
```diff
@@ -54,57 +58,106 @@ const getNetwork = async () => {
...
+		return {
+		  raw,
+		  txMetadata: merkleizedMetadata.getProofForExtrinsicPayload(u8aToHex(raw.toU8a(true)))
+		};
+	  }
+
+	public async signPayload(payload: SignerPayloadJSON): Promise<SignerResult> {
+
+		const { address } = await this.#getLedger.getAddress(
+			Number(this.#api.consts.system.ss58Prefix),
```
Comment (TarikGul) [inline blockchainServices/setup.ts:116]: I would stay away from using `Number` as it is pretty un performant

**Review by TarikGul — CHANGES_REQUESTED** [review]
Few nits, overall really good!

File: blockchainServices/setup.ts [hunk 3]
```diff
@@ -54,57 +58,106 @@ const getNetwork = async () => {
 }
 
 const getLedgerAddress = async () => {
+	const api = await getApi();
 	const network = await getNetwork();
-	const ledger = new Ledger('hid', network);
-	return await ledger.getAddress();
+	const ledger = new LedgerGeneric('hid', network, knownLedger['polkadot']);
```
[the remark this answers is not in the document]
Comment (bee344) [reply blockchainServices/setup.ts]: Nice catch, solved now

**Review by TarikGul — APPROVED** [review]
Great job!

**Pull request merged**